### PR TITLE
Windows icon fix and change the installer to QT Installer Framework on Windows

### PR DESCRIPTION
**Pull request by MTRNord**
This PR 1. Adds the missing AppIcon to windows and removes the console too. Code is taken from https://github.com/euler0/mini-cmake-qt

I moved ${OS_BUNDLE} as it needs to always be the first argument after the project name.

Debug Console on windows needs to be enabled by setting the Cmake build type to not be "Release"

Feel free to squash merge them ;) I used the web interface at the beginning

### CODE REVIEW

File: .ci/windows/nheko.nsi [hunk 1]
```diff
@@ -1,225 +0,0 @@
-############################################################################################
-#      NSIS Installation Script created by NSIS Quick Setup Script Generator v1.09.18
-#               Entirely Edited with NullSoft Scriptable Installation System
-#              by Vlasis K. Barkas aka Red Wine red_wine@freemail.gr Sep 2006
-############################################################################################
-
-!define APP_NAME "nheko"
-!define COMP_NAME "mujx"
```
Comment (mujx) [inline .ci/windows/nheko.nsi]: Use nheko here.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
+# Include Qt basic functions
+include(QtCommon)
+
+project(nheko LANGUAGES C CXX VERSION 0.0.1)
+
+# Set PROJECT_VERSION_PATCH and PROJECT_VERSION_TWEAK to 0 if not present, needed by add_project_meta
+fix_project_version()
+
+# Set additional project information
+set(COMPANY "Mujx")
```
Comment (mujx) [inline CMakeLists.txt]: Also nheko here.

File: cmake/MacOSXBundleInfo.plist.in [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    <string>6.0</string>
+    <key>CFBundleLongVersionString</key>
+    <string>${MACOSX_BUNDLE_LONG_VERSION_STRING}</string>
+    <key>CFBundleName</key>
+    <string>${MACOSX_BUNDLE_BUNDLE_NAME}</string>
+    <key>CFBundlePackageType</key>
+    <string>APPL</string>
+    <key>CFBundleShortVersionString</key>
+    <string>${MACOSX_BUNDLE_SHORT_VERSION_STRING}</string>
+    <key>CFBundleSignature</key>
```
Comment (mujx) [inline cmake/MacOSXBundleInfo.plist.in]: You can remove this key and the tag below.

File: cmake/QtCommon.cmake [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+macro(fix_win_compiler)
+if (MSVC)
+    set_target_properties(${PROJECT_NAME} PROPERTIES
+        WIN32_EXECUTABLE YES
+        LINK_FLAGS "/ENTRY:mainCRTStartup"
+    )
+endif()
+endmacro()
+
+macro(init_qt)
```
Comment (mujx) [inline cmake/QtCommon.cmake]: You can remove this macro and the `init_qt()` call below. It's already handled.

File: appveyor.yml [hunk 5]
```diff
@@ -15,14 +15,34 @@ build:
...
     - set PATH=%PATH%;%QT_DIR%\bin;C:\MinGW\bin
-    - copy LICENSE .ci\windows\LICENSE.txt
-    - cinst nsis --version 3.0 -y -installArgs /D=C:\nsis
 
 build_script:
-    - cmake -G "Visual Studio 14 2015 Win64" -H. -Bbuild -DAPPVEYOR_BUILD=ON
+    # VERSION format:     branch-master/branch-1.2
+    # INSTVERSION format: x.y.z
+    # WINVERSION format:  9999.0.0.123/1.2.0.234
+    - if "%APPVEYOR_REPO_TAG%"=="false" set INSTVERSION=0.0.1
```
Comment (mujx) [inline appveyor.yml]: The version is 0.1.0

File: .ci/windows/nheko.nsi (hunk 1 above)
Comment (MTRNord) [reply .ci/windows/nheko.nsi]: This is the file which got removed :)

File: cmake/MacOSXBundleInfo.plist.in (hunk 3 above)
Comment (MTRNord) [reply cmake/MacOSXBundleInfo.plist.in]: Which tag do you mean? `CFBundleVersion` or the `<string>`?

File: .ci/windows/nheko.nsi (hunk 1 above)
Comment (mujx) [reply .ci/windows/nheko.nsi]: Oh right 😅

File: cmake/MacOSXBundleInfo.plist.in (hunk 3 above)
Comment (mujx) [reply cmake/MacOSXBundleInfo.plist.in]: The string that goes together with `CFBundleSignature`

**Comment by MTRNord** [comment]
Todo: prevent the cleanup from crashing

**Comment by PureTryOut** [comment]
@MTRNord what exactly do you still have left to do before you think this is ready for merging? Besides fixing the conflict in CMakeLists.txt of course.

**Comment by mujx** [comment]
@MTRNord If fixing the cleanup crash is the blocker here, we can skip it for now and merge the PR with just the new qt installer.

**Comment by MTRNord** [comment]
@mujx I cant work on this PR before tommorow. But I can make it ready

**Comment by MTRNord** [comment]
Todo: Fix cleanup. (Done in a new PR)

**Pull request merged**